feat: Added overview page to display pokemons

ChristianAsilo opened the pull request:
- Removed home page
- Added pokemon overview folder
- Added PokemonOverviewVm class
- Added PokemonOverviewConnector
- Added PokemonOverviewPage to display pokemons
- Added new constants
![display](https://user-images.githubusercontent.com/128013689/227496377-96012df1-23d8-4efa-a163-92c75c23a899.PNG)
![analyze](https://user-images.githubusercontent.com/128013689/227172348-e9658931-99b3-46f9-b904-8353923e40df.PNG)
![dartFormat](https://user-images.githubusercontent.com/128013689/227172355-daedd562-fa84-4803-ab4f-3d247d4e207b.PNG)

chrstn06 committed:
feat: Added overview page to display pokemons
- Removed home page
- Added pokemon overview folder
- Added PokemonOverviewVm class
- Added PokemonOverviewConnector
- Added PokemonOverviewPage to display pokemons
- Added new constants

chrstn06 committed:
feat: Added overview page to display pokemons
- Added pokemon card for display
- Added dartx in pubspec.yaml
- Removed the use of index for getting pokemon id
- Modified the pokemon image url

shawn-ffufm commented on lib/features/pokemon_overview/pokemon_overview_page.dart:
```diff
@@ -0,0 +1,36 @@
+
+  @override
+  Widget build(BuildContext context) {
+    pokemons;
```
do u still need this here? 😅

shawn-ffufm commented on lib/utils/constants.dart:
```diff
@@ -1,4 +1,16 @@
+//Pokemon Overview Page
+const appBarTitle = 'Pokedex';
+const gridCount = 2;
+const defaultOne = 1;
```
do u still need this?

shawn-ffufm commented on lib/utils/constants.dart:
```diff
@@ -1,4 +1,16 @@
+const appBarTitle = 'Pokedex';
+const gridCount = 2;
+const defaultOne = 1;
+const cardPadding = EdgeInsets.all(20);
```
seems this is no need, just directly use its value to where it's used 🙂

shawn-ffufm commented on lib/utils/constants.dart:
```diff
@@ -1,4 +1,16 @@
+const gridCount = 2;
+const defaultOne = 1;
+const cardPadding = EdgeInsets.all(20);
+const crossAlignmentCenter = CrossAxisAlignment.center;
```
same here

shawn-ffufm commented on lib/utils/constants.dart:
```diff
@@ -1,4 +1,16 @@
+const defaultOne = 1;
+const cardPadding = EdgeInsets.all(20);
+const crossAlignmentCenter = CrossAxisAlignment.center;
+const textSize = TextStyle(fontSize: 20);
```
same here

shawn-ffufm commented on lib/utils/constants.dart:
```diff
@@ -1,4 +1,16 @@
+const crossAlignmentCenter = CrossAxisAlignment.center;
+const textSize = TextStyle(fontSize: 20);
+const imageLink = 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/home/[index].png';
+const indexString = '[index]';
```
rename this to `indexKey` 🙂

shawn-ffufm commented on lib/widget/pokemon_card.dart:
```diff
@@ -0,0 +1,34 @@
+
+class PokemonCard extends StatelessWidget {
+  const PokemonCard({
+    required this.pokemonName,
```
since pokemon model have only 2 properties, why not just pass the whole model inside your pokemon card instead? 🙂

shawn-ffufm reviewed (changes requested):
see pr feedback 🙂

chrstn06 committed:
feat: Added overview page to display pokemons
(pr feedback)
- Removed unnecessary constants
- Renamed indexString into indexKey
- Passed the whole Pokemon model in pokemonCard
